### DISCUSSION TITLE
Have a problem in XFEM

### DISCUSSION THREAD

**Discussion by yf-comon** [idaholab/moose#28341] [Q&A Modules: Solid mechanics]
When calculating the crack, I don't know why the following error occurs. How should I solve it?
```
*** ERROR ***
Attemping to add intersection to an invalid fragment edge. Element: 4019 fragment_edge: 4
```

**Comment by GiudGiud** [comment]
@hugary1995

**Comment by jiangwen84** [comment]
tbis is not a trival error. it means the cut algorithm fails to add a cut to one element. it can be caused by multiple cuts, or some wired cutting locations.

**Reply by yf-comon** [reply]
Should I reduce the time step at this point? I would like to know what I should do next to fix this

**Reply by GiudGiud** [reply]
when does this appear? At initialization or on the first time step or later?

**Reply by yf-comon** [reply]
It happened after I calculated it for a while.

**Reply by GiudGiud** [reply]
ok use a checkpoint ouptut and restart (see this: https://mooseframework.inl.gov/source/outputs/Checkpoint.html) maybe a few time steps before the problem occurs. 
Then try to fiddle with the time step or solve parameters.

If this does not work, then actually maybe re-mesh the geometry?
@jiangwen84 what do you think?

**Reply by yf-comon** [reply]
I tried to reduce the time step, but the question still arises, should I go to the encrypted mesh?

**Reply by GiudGiud** [reply]
what do you mean by encrypted?

**Reply by jiangwen84** [reply]
xfem does not support checkpoint. share with us your input file and I might be able to take a look.

**Reply by yf-comon** [reply]
Add more nodes to make the grid smaller and thinner

**Reply by yf-comon** [reply]
[input.zip](https://github.com/user-attachments/files/16556233/input.zip)
I compressed the sub-card into an installation package, where the grid of the card is a cross-section of the wall of a pipe, divided into three layers. I transferred the temperature and displacement to the sub-card in the main card in 3D, and the settings of the XFEM part are very I show below：
```
[XFEM]
    geometric_cut_userobjects = 'cut_mesh'
    qrule = volfrac
    output_cut_plane = true
  []
  [DomainIntegral]
    integrals = 'Jintegral InteractionIntegralKI InteractionIntegralKII'
    displacements = 'disp_x disp_y'
    crack_front_points_provider = cut_mesh
    2d=true
    number_points_from_provider = 0
    crack_direction_method = CurvedCrackFront
    radius_inner = '0.01'
    radius_outer = '0.03'
    youngs_modulus = 207000
    poissons_ratio = 0.3
    #block = 'TC'
    incremental = true
    used_by_xfem_to_grow_crack = true
  []


  [UserObjects]
    inactive = 'line_seg_cut_uo xfem_marker_uo'
    [./line_seg_cut_uo]
      type = LineSegmentCutSetUserObject
      cut_data = '-2e-4   -2e-04   -1.75e-04 -2.025e-04  0.0  3600
                  -1.6e-4   -2.03e-04   -1.5e-04 -2.05e-04  5000 7200
                  -1.45e-4   -2.06e-04   -1.2e-04 -2.03e-04  7200  10800
                  -1.2e-4   -2.03e-04   -1e-04 -2e-04  10800  14400
                  -1e-4   -1.5e-04   0      -2e-04  14400   144000'
    [../]
    [./xfem_marker_uo]
      type = XFEMRankTwoTensorMarkerUserObject
      execute_on = timestep_end
      tensor = stress
      scalar_type = MaxPrincipal
      threshold = strength
      average = true
    [../]
    [nucleate]
      type = MeshCut2DRankTwoTensorNucleation
      tensor = stress
      #scalar_type = VonMisesStress
      #initiate_on_boundary = tgo_tc
      scalar_type = MaxPrincipal
      nucleation_threshold = strength
      nucleation_length = 0.001
      #edge_extension_factor = 2e-5
      nucleate_across_full_element = true
      []
    [cut_mesh]
      type = MeshCut2DFractureUserObject
      mesh_file = make_edge_crack_in.e
      k_critical= 80
      growth_increment = 0.001
      nucleate_uo = nucleate
    []

  []
```
